src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -220,6 +222,13 @@ private void validateTable(TableModel table, Errors errors, SchemaValidationCont
         }
     }
 
+    private void validateDatatype(ColumnModel column, Errors errors) {
+        if (!VALID_DATATYPES.contains(column.getDatatype().toLowerCase())){
+            errors.rejectValue("datatype", "InvalidDatatype",
+                "invalid datatype " + column.getDatatype() + " valid datatypes are " + String.join(",", VALID_DATATYPES.asList()));
```

Comment:
Is `asList()` needed here? `String.join()` accepts `Iterable`
```suggestion
                "invalid datatype " + column.getDatatype() + " valid datatypes are " + String.join(",", VALID_DATATYPES));
```